hoomd/mpcd/ReverseNonequilibriumShearFlowUtilities.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file ReverseNonequilibriumShearFlowUtilities.h
+ * \brief Helper functions for Reverse nonequilibrium shear flow
+ */
+
+#ifndef MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_UTILITIES_H_
+#define MPCD_REVERSE_NONEQUILIBRIUM_SHEAR_FLOW_UTILITIES_H_
+
+#ifdef NVCC
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#endif
+
+#include "hoomd/HOOMDMath.h"
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+namespace detail
+    {
+
+//! Custom comparison operator for sorting arrays by their momentum closest to a target value.
+class ReverseNonequilibriumShearFlowSorter
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param p_target Target momentum
+     */
+    HOSTDEVICE ReverseNonequilibriumShearFlowSorter(Scalar p_target) : p(p_target) { }
+
+    //! Sorts two momenta
+    /*!
+     * \param in0 Momentum of particle 0
+     * \param in1 Momentum of particle 1
+     *
+     * \returns True if \a in0 has momentum closer to the target value than \a in1
+     *
+     * Particles to be sorted \b must contain only the values with the appropriate
+     * sign. (Particles to sort in the top slab have all negative momentum,
+     * particles in the bottom slab have all positive momentum.)
+     *
+     * Particles in the top slab get sorted according to the closest value to -p,
+     * and particles in the bottom slab get sorted according to the closest value to +p.
```

Comment:
This is a detail of use elsewhere, not this class, so we wouldn't usually document it here.
```suggestion
```